samples: matter: Add support for non-PM builds (excluding nrf5340/nrf9160)

nordicjm opened the pull request:
Includes configuration for matter samples also

Blocked by NCSDK-36367 and NCSDK-36368

NordicBuilder commented:
The following west manifest projects have changed revision in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zephyr | https://github.com/nrfconnect/sdk-zephyr/commit/9a699b5802cbaac098cca7c0ece52f7f862504a4 | https://github.com/nrfconnect/sdk-zephyr/commit/f29cbf9a48a032237a3508d58a3570679d36667f| [nrfconnect/sdk-zephyr@9a699b58..f29cbf9a](https://github.com/nrfconnect/sdk-zephyr/compare/9a699b5802cbaac098cca7c0ece52f7f862504a4..f29cbf9a48a032237a3508d58a3570679d36667f) |

✅ **All manifest checks OK**

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

nordicjm committed:
manifest: Update sdk-connectedhomeip

Includes changes for partition removal

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: matter: Update samples to use non-PM board targets

Includes nrf52840dk, nrf54l15dk (secure only) and nrf54lm20dk

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
dts: Add common board files with partitioning for matter

Adds dtsi files which can be included and set the partition layout
up for the matter samples

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: matter: kconfig: Exclude disabling PM for non-secure builds

TF-M is currently not supported without partition manager, do not
force disabling PM if a non-secure board target is used in samples
that build for these

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

ArekBalysNordic committed:
samples: matter: Add CHIP_APP_ZAP_DIR kconfig to the mfg sample

We need to use the new CONFIG_CHIP_APP_ZAP_DIR to set the proper
path to the custom zap-generated directory in Matter stack.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

nordicjm committed:
samples: matter: light_bulb: cmake: Fix file path

Fixes an invalid path to a source file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
sysbuild: matter: Migrate factory data from partition manager

Allows generating and flashing the nrf70 firmware in applications
which do not use partition manager, and will be flashed when west
flash is invoked

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
cmake: Add hex image flashing application

This adds an image which sysbuild can add which simply flashes a
hex file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented on applications/matter_factory_data/CMakeLists.txt:
```diff
@@ -0,0 +1,53 @@
+
+# Do not apply shield or snippets for the factory data partition and use an empty file for the
+# board defconfig file
+set(BOARD_DEFCONFIG ${CMAKE_CURRENT_SOURCE_DIR}/prj.conf)
```
> This approach makes no sense. There are only weird hacks and overrides for the process that relies only on CMAKE.

You're not a build system maintainer, these are not hacks.

> I don't see any advantage in creating a separate image for factory data. This only creates many redundant files in the build directory.
> 
> Instead align this PR to this approach: https://github.com/nrfconnect/sdk-nrf/pull/26394

This is the approach and this is how it is going to be done, just like it is for other images and other systems e.g. in zephyr, in bare metal, etc.

ArekBalysNordic reviewed (changes requested):
Detected the following inconsistencies:

1. https://github.com/nrfconnect/sdk-nrf/blob/61d93cb1804e776812ba1eda60fdf11630519e9d/samples/matter/template/sysbuild/mcuboot/boards/nrf54lm20dk_nrf54lm20a_cpuapp_internal.conf#L23
2. https://github.com/nrfconnect/sdk-nrf/blob/61d93cb1804e776812ba1eda60fdf11630519e9d/samples/matter/template/sysbuild/mcuboot/boards/nrf54lm20dk_nrf54lm20a_cpuapp.conf#L18
3. https://github.com/nrfconnect/sdk-nrf/blob/61d93cb1804e776812ba1eda60fdf11630519e9d/samples/matter/template/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l15_cpuapp.overlay#L14
